hack/generator/pkg/astbuilder/assignments.go

```diff
@@ -6,26 +6,59 @@
 package astbuilder
 
 import (
+	"fmt"
 	"go/token"
 
 	"github.com/dave/dst"
 )
 
 // SimpleAssignment performs a simple assignment like:
-//     <lhs> := <rhs>       // tok = token.DEFINE
-// or  <lhs> = <rhs>        // tok = token.ASSIGN
-func SimpleAssignment(lhs dst.Expr, tok token.Token, rhs dst.Expr) *dst.AssignStmt {
+//
+//     <lhs> = <rhs>
+//
+func SimpleAssignment(lhs dst.Expr, rhs dst.Expr) *dst.AssignStmt {
 	return &dst.AssignStmt{
 		Lhs: []dst.Expr{
 			dst.Clone(lhs).(dst.Expr),
 		},
-		Tok: tok,
+		Tok: token.ASSIGN,
 		Rhs: []dst.Expr{
 			dst.Clone(rhs).(dst.Expr),
 		},
 	}
 }
 
+// SimpleDeclaration performs a simple assignment like:
+//
+//     <id> := <rhs>
+//
+func SimpleDeclaration(id string, rhs dst.Expr) *dst.AssignStmt {
+	return &dst.AssignStmt{
+		Lhs: []dst.Expr{
+			dst.NewIdent(id),
+		},
+		Tok: token.DEFINE,
+		Rhs: []dst.Expr{
+			dst.Clone(rhs).(dst.Expr),
+		},
+	}
+}
+
+// SetVariable allows for either variable declaration or assignment by passing the required token
+// Only token.DEFINE and token.ASSIGN are supported, other values will panic.
+// Use SimpleAssignment or SimpleDeclaration by preference
+func SetVariable(lhs dst.Expr, tok token.Token, rhs dst.Expr) *dst.AssignStmt {
```

Comment:
Minor: Rather than `SetVariable` this could possibly just be `Assign` (or `Assignment`) in contrast to `SimpleAssignment`/`SimpleDeclaration`?

That keeps with the nomenclature of the `dst` and so hopefully makes this easier to find and more closely matches the naming of the above two methods (which I think fits given that this is "like those but with more flexibility and less safety")?

Reply:
When I hear variable I tend to think of local (or global) variables and not expressions like `a.b.c.d` too, whereas this supports both

Reply:
Renamed to `AssignmentStatement()`.